File: inc.lib/classes/MagicAdmin/Entity/Data/MessageFolder.php
<?php

namespace MagicAdmin\Entity\Data;

use MagicObject\MagicObject;

/**
 * The MessageFolder class represents an entity in the "message_folder" table.
 *
 * This entity maps to the "message_folder" table in the database and supports ORM (Object-Relational Mapping) operations. 
 * You can establish relationships with other entities using the JoinColumn annotation. 
 * Ensure to include the appropriate "use" statement if related entities are defined in a different namespace.
 * 
 * For detailed guidance on using the MagicObject ORM, refer to the official tutorial:
 * @link https://github.com/Planetbiru/MagicObject/blob/main/tutorial.md#orm
 * 
 * @package MagicAdmin\Entity\Data
 * @Entity
 * @JSON(property-naming-strategy=SNAKE_CASE, prettify=false)
 * @Table(name="message_folder")
 */
class MessageFolder extends MagicObject
{
	/**
	 * Message Folder ID
	 * 
	 * @Id
	 * @GeneratedValue(strategy=GenerationType.UUID)
	 * @Column(name="message_folder_id", type="varchar(40)", length=40, nullable=true)
	 * @Label(content="Message Folder ID")
	 * @var string
	 */
	protected $messageFolderId;

	/**
	 * Name
	 * 
	 * @Column(name="name", type="varchar(100)", length=100, nullable=true)
	 * @Label(content="Name")
	 * @var string
	 */
	protected $name;

	/**
	 * Sort Order
	 * 
	 * @NotNull
	 * @Column(name="sort_order", type="int", nullable=false)
	 * @Label(content="Sort Order")
	 * @var int
	 */
	protected $sortOrder;

	/**
	 * Time Create
	 * 
	 * @Column(name="time_create", type="timestamp", length=19, nullable=true, updatable=false)
	 * @Label(content="Time Create")
	 * @var string
	 */
	protected $timeCreate;

	/**
	 * Time Edit
	 * 
	 * @Column(name="time_edit", type="timestamp", length=19, nullable=true)
	 * @Label(content="Time Edit")
	 * @var string
	 */
	protected $timeEdit;

	/**
	 * Admin Create
	 * 
	 * @Column(name="admin_create", type="varchar(40)", length=40, nullable=true, updatable=false)
	 * @Label(content="Admin Create")
	 * @var string
	 */
	protected $adminCreate;

	/**
	 * Admin Edit
	 * 
	 * @Column(name="admin_edit", type="varchar(40)", length=40, nullable=true)
	 * @Label(content="Admin Edit")
	 * @var string
	 */
	protected $adminEdit;

	/**
	 * IP Create
	 * 
	 * @Column(name="ip_create", type="varchar(50)", length=50, nullable=true, updatable=false)
	 * @Label(content="IP Create")
	 * @var string
	 */
	protected $ipCreate;

	/**
	 * IP Edit
	 * 
	 * @Column(name="ip_edit", type="varchar(50)", length=50, nullable=true)
	 * @Label(content="IP Edit")
	 * @var string
	 */
	protected $ipEdit;

	/**
	 * Active
	 * 
	 * @NotNull
	 * @Column(name="active", type="tinyint(1)", length=1, defaultValue="TRUE", nullable=false)
	 * @DefaultColumn(value="TRUE")
	 * @Label(content="Active")
	 * @var bool
	 */
	protected $active;

}

File: inc.graphql-resources/backend/kotlin/model/entity/core/MessageFolder.kt
package com.planetbiru.graphqlapplication.model.entity.core

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "message_folder")
data class MessageFolder(
    @Id
    @Column(name = "message_folder_id", nullable = false, length = 40)
    var messageFolderId: String,

    @Column(name = "name", length = 100)
    var name: String? = null,

    @Column(name = "admin_id", length = 40)
    var adminId: String? = null,

    @Column(name = "sort_order")
    var sortOrder: Int? = null,

    @Column(name = "time_create")
    var timeCreate: LocalDateTime? = null,

    @Column(name = "time_edit")
    var timeEdit: LocalDateTime? = null,

    @Column(name = "admin_create", length = 40)
    var adminCreate: String? = null,

    @Column(name = "admin_edit", length = 40)
    var adminEdit: String? = null,

    @Column(name = "ip_create", length = 50)
    var ipCreate: String? = null,

    @Column(name = "ip_edit", length = 50)
    var ipEdit: String? = null,

    @Column(name = "active")
    var active: Boolean? = true,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "admin_id", referencedColumnName = "admin_id", insertable = false, updatable = false)
    val admin: Admin? = null
)